### PR TITLE
changing the onnxwrapper script for gpu issue

### PR DESCRIPTION
**Pull request by sanjaypavo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
@sanjaypavo could you please describe your motivation in this PR?

File: mmdeploy/backend/onnxruntime/wrapper.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ def __init__(self,
             not exist: {ort_custom_op_path}')
         device_id = parse_device_id(device)
         is_cuda_available = ort.get_device() == 'GPU'
```
Comment (RunningLeon) [inline mmdeploy/backend/onnxruntime/wrapper.py:53]: may remove this line and add
```
if device != 'cpu':
    assert ort.get_device() == 'GPU', f'Should `pip install onnxruntime-gpu` if you want to run on {device} device.' 
```

File: mmdeploy/backend/onnxruntime/wrapper.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def __init__(self,
         self.io_binding = sess.io_binding()
         self.device_id = device_id
         self.is_cuda_available = is_cuda_available
```
Comment (RunningLeon) [inline mmdeploy/backend/onnxruntime/wrapper.py:63]: may remove this line.

**Comment by sanjaypavo** [comment]
> @sanjaypavo could you please describe your motivation in this PR?

To be able to work with cpu and gpu with the same [GPU docker](https://github.com/open-mmlab/mmdeploy/blob/master/docker/GPU/Dockerfile)

**Comment by RunningLeon** [comment]
@sanjaypavo Hi, if possible, could you kindly update the PR based on the comments?

**Comment by sanjaypavo** [comment]
> @sanjaypavo Hi, if possible, could you kindly update the PR based on the comments?

Hi, Updated and merged the changes.

**Review by grimoire — APPROVED** [review]

File: mmdeploy/backend/onnxruntime/wrapper.py (hunk 2 above)
Comment (RunningLeon) [reply mmdeploy/backend/onnxruntime/wrapper.py:64]: @sanjaypavo Hi, could you remove the unused attribute `is_cuda_available`?

Comment (sanjaypavo) [reply mmdeploy/backend/onnxruntime/wrapper.py]: > @sanjaypavo Hi, could you remove the unused attribute `is_cuda_available`?

@RunningLeon Hi, remove the attribute.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @sanjaypavo ！First of all, we want to express our gratitude for your significant PR in the mmdeploy project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**